tap_freshdesk/utils.py

```diff
@@ -56,12 +56,18 @@ def load_schema(entity):
     return load_json(get_abs_path("schemas/{}.json".format(entity)))
 
 
-def update_state(state, entity, dt):
+def update_state(state, entity, dt, query_dt=None):
     if dt is None:
         return
 
     if isinstance(dt, datetime.datetime):
         dt = strftime(dt)
+    
+    if query_dt:
+        if isinstance(query_dt, datetime.datetime):
+            query_dt = strftime(query_dt)
```

Comment:
you want to compare strings? doesnt make sense 
Also you should rename the variables to query_str, query_dt to 1. not mutate 2. easier to read code

Reply:
I can rename the variable.

The reason this was cast first was because the other date is also cast first. Maybe there's a reason for that (like the dates coming in from freshdesk are strings).

Reply:
Actually looking at it again I think the variable name makes sense (since it's being passed in as a datetime). It gets cast into a string for comparison because everything else is a string. Let me know if that's what you meant or if you have other suggestions.